torch/csrc/jit/script/parser.h

```diff
@@ -29,6 +29,20 @@ struct Parser {
         List<Expr>(makeList(range, std::move(inputs))),
         List<Attribute>(makeList(range, std::move(attributes))));
   }
+  // exp | expr, | expr, expr, ...
+  TreeRef parseExpOrExpList(int end) {
+    auto prefix = parseExp();
+    if(L.cur().kind == ',') {
+      std::vector<Expr> exprs = { prefix };
+      while(L.cur().kind != end) {
+        L.expect(',');
+        exprs.push_back(parseExp());
+      }
+      auto list = List<Expr>::create(prefix.range(), exprs);
+      prefix = ListLiteral::create(list.range(), list);
+    }
+    return prefix;
+  }
```

Comment:
Isn't that just `parseList` with `TK_NOTHING` as beginning and end?

Reply:
No. In the single-expression case it returns a single  `Expr`. In the multi-expression case it returns a `ListLiteral`. parseList always returns a `List<Expr>`